src/main/kotlin/net/atos/zac/productaanvraag/ProductaanvraagService.kt

```diff
@@ -180,101 +180,177 @@ class ProductaanvraagService @Inject constructor(
     ) {
         var initiatorAdded = false
         productaanvraag.betrokkenen?.forEach {
-            when (it.rolOmschrijvingGeneriek) {
-                Betrokkene.RolOmschrijvingGeneriek.ADVISEUR -> {
-                    addBetrokkene(it, OmschrijvingGeneriekEnum.ADVISEUR, zaak)
-                }
+            initiatorAdded = if (it.roltypeOmschrijving != null) {
+                addBetrokkenenWithRole(it, initiatorAdded, zaak)
+            } else {
+                addBetrokkenenWithGenericRole(it, initiatorAdded, zaak)
+            }
+        }
+    }
 
-                Betrokkene.RolOmschrijvingGeneriek.BELANGHEBBENDE -> {
-                    addBetrokkene(it, OmschrijvingGeneriekEnum.BELANGHEBBENDE, zaak)
+    private fun addBetrokkenenWithRole(
+        betrokkene: Betrokkene,
+        initiatorAdded: Boolean,
+        zaak: Zaak
+    ): Boolean {
+        when (betrokkene.roltypeOmschrijving.lowercase()) {
```

Comment:
Can't we improve this `lowercase()` and `toString()` comparison? Looks quite messy..

Reply:
Changed it to use `equals(ignoreCase=true)` cause we're comparing the role description value against existing enum constants.

Perhaps we need a dedicated string constants here

Reply:
ok. good enough for now I think

Reply:
Using dedicated constants now, although relying on description feels a bit fragile